docs/rfc/0002-plugins.md

```diff
@@ -0,0 +1,184 @@
+[Rego](https://www.openpolicyagent.org/docs/latest/policy-language/)
+
+## Motivation
+
+Let's take the helm `Renderer` example: the community is producing some
+quality charts users would like to be able to reuse.
+
+- It is counterproductive to start translating complex helm charts in `js`
+instead of reusing them.
+- It is often necessary to modify the Helm Chart Kubernetes objects in ways
+the original authors haven't thought of. Importing them in `jk` allows just
+that.
+
+## Design
+
+### General Flow
+
+A call to a plugin a really an RPC call issued to a plugin server process.
+`jk` is responsible for the life cycle of those plugin processes.
+
+From an API point of view, it is envisioned that library would provide nice
+wrapper objects. For instance the helm chart renderer could look like:
+
+```js
+const redis = new k8s.helm.Chart("redis", {
+    repo: "stable",
+    chart: "redis",
+    version: "3.10.0",
+    values: {
+        usePassword: true,
+        rbac: { create: true },
+    },
+ });
+
+redis.render().then(std.log);
+```
+
+The `Chart` object, in this case, would be part of the `@jkcfg/kubernetes` library.
+
+The `render()` function of the `Chart` is implemented with a standard library
+RPC call that lands in the go part of `jk`. Then:
+
+- `jk` checks if a helm renderer plugin is already running and spawns a new one if needed
+- `jk` waits until the plugin process has started and is ready to accept RPCs.
+- `jk` issues a RPC call to the plugin process
+- The plugin process returns an answer
+- The answer is serialized and sent back to the `js` vm
+
+### The standard library `plugin` function
+
+The core `plugin` RPC call is quite general:
+
+```text
+plugin(kind: string, url: string, input: JSON) -> JSON
+```
+
+- `kind` is the kind of plugin invoked (eg. `render`). Plugin binaries can
+implement more than one kind of plugins.
+- `url` identifies a plugin, for instance:
+`https://jkcfg.github.io/plugins/helm/0.1.0/plugin.json`. This JSON file is
+really plugin metadata (see next section). It is highly recommended, for
+reproducibility, to ensure the plugin definition and binaries the URL points
+to be immutable to encode a version in the URL.
+- `input` and the return value are generic JSON objects that wrapping library
+code are responsible for understanding.
+
+### Plugin definition
+
+The plugin URL in the `plugin` call points to a JSON file describing the plugin:
+
+```json
+{
+  "name": "helm",
+  "version": "0.1.0",
+  "binaries": {
+    "linux-amd64": "https://jkcfg.github.io/plugins/helm/0.1.0/linux-amd64/jk-plugin-helm",
+    "darwin-amd64": "https://jkcfg.github.io/plugins/helm/0.1.0/darwin-amd64/jk-plugin-helm"
+  }
+}
+```
+
+### Local plugins
+
+For development purposes it is possible to point the `plugin` RPC call to a
+local file by giving a relative path to the JSON plugin definition. The
+`binaries` fields can point at plugins present in the `PATH`.
```

Comment:
Another formulation of this would be to expect a relative path to the binary. This would make sense if the binaries are shipped with the code wrapping them, for example.